source/tests-integration/TestFixture.ts

```diff
@@ -0,0 +1,91 @@
+        this.configuration = configuration;
+        this.connector = connector;
+        this.accountManager = accountManager;
+        this.contractDeployer = contractDeployer
+    }
+
+    public static create = async (): Promise<TestFixture> => {
+        const configuration = await Configuration.create();
+        await TestRpc.startTestRpcIfNecessary(configuration);
+        const compiledContracts = await new ContractCompiler(configuration).compileContracts();
+        const connector = new Connector(configuration);
+        const accountManager = new AccountManager(configuration, connector);
+        const contractDeployer = new ContractDeployer(configuration, connector, accountManager, compiledContracts);
+        await contractDeployer.deploy();
+        return new TestFixture(configuration, connector, accountManager, contractDeployer);
+    }
+
+    public async approveCentralAuthority(): Promise<void> {
+        const authority = this.contractDeployer.getContract('Augur');
+        const cash = <Cash>this.contractDeployer.getContract('Cash');
+        const transactionHash = await cash.approve(authority.address, new BN(2).pow(new BN(256)).sub(new BN(1)));
+        await this.connector.waitForTransactionReceipt(transactionHash, `Approving central authority.`);
+    }
+
+    public async createMarket(universe: Universe, numOutcomes: number, endTime: number, feePerEthInWei: number, denominationToken: string, designatedReporter: string, numTicks: number): Promise<Market> {
+        const legacyReputationToken = <LegacyReputationToken>this.contractDeployer.getContract('LegacyReputationToken');
+        const reputationTokenAddress = await universe.getReputationToken_();
+        const reputationToken = new ReputationToken(this.connector, this.accountManager, reputationTokenAddress, this.configuration.gasPrice);
+
+        // get some REP
+        // TODO: just get enough REP to cover the bonds rather than over-allocating
+        const repFaucetTransactionHash = await legacyReputationToken.faucet(new BN(0));
+        const repApprovalTransactionHash = await legacyReputationToken.approve(reputationTokenAddress, new BN(2).pow(new BN(256)).sub(new BN(1)));
+        await this.connector.waitForTransactionReceipt(repFaucetTransactionHash, `Using legacy reputation faucet.`);
+        await this.connector.waitForTransactionReceipt(repApprovalTransactionHash, `Approving legacy reputation.`);
+        const repMigrationTransactionHash = await reputationToken.migrateFromLegacyReputationToken();
+        // necessary because it is used part of market creation fee calculation
+        const currentReportingWindowTransactionHash = await universe.getCurrentReportingWindow();
+        // necessary because it is used as part of market creation fee calculation
+        const previousReportingWindowTransactionHash = await universe.getPreviousReportingWindow();
```

Comment:
Mostly just curious about this. Would you also be able to just call `universe.getMarketCreationCost()` instead of `universe.getMarketCreationCost_()` and these prep calls?

Reply:
Calling a remote function does not give you the return value.  Calling a local function does not cause it to mutate storage.  For some functions, we need to both mutate storage and get the return value, which is why we have to do _both_ a local call and a remote call.  See the giant discussion in #dev filled with @tinybike  😡 for details on why this is.  😄

Reply:
Ah yea that makes sense now reading it.